BatchedMesh: Refactor error handling.

Makio64 opened the pull request:
Fixed https://github.com/mrdoob/three.js/issues/29379

**Description**

Replace "silent error handling" by throwing clear error message to help the devs to find errors early when using BatchedMesh.

Makio64 committed:
Refactor BatchedMesh class error handling

Makio64 committed:
Refactor BatchedMesh class error handling

Makio64 committed:
fix linting

Mugen87 committed:
GTAONode: Fix AO with WebGL backend. (#29593)

* GTAONode: Fix AO with WebGL backend.

* E2E: Update screenshot.

* DenoiseNode: Use `getViewPosition()` from core.

* DenoiseNode: Clean up.

sunag committed:
OutlineNode: Improve approach (#29583)

* improve outline example

* rev

sunag committed:
WebGPURenderer: Introduce `PostProcessingUtils` (#29595)

mrdoob committed:
Editor: Fixed editor not loading on Chrome 129.0.6668.100 due to import maps definitions location.

gkjohnson commented on src/objects/BatchedMesh.js:
```diff
@@ -651,7 +651,7 @@ class BatchedMesh extends Mesh {
 		if ( geometryId >= drawRanges.length || drawRanges[ geometryId ].active === false ) {
 
-			return this;
+			throw new Error( `BatchedMesh: Invalid geometryId ${geometryId}. Geometry is either out of range or inactive.` );
```
We should say that the geometry is "deleted" rather than "inactive". The "active" state isn't a concept exposed to end users.

Makio64 replied:
done !

aardgoose committed:
remove obsolete method (#29596)

Co-authored-by: aardgoose <angus.sawyer@email.com>

Mugen87 committed:
Addons: Add `SSRNode`. (#29597)

* Addons: Add `SSRNode`.

* E2E: Update screenshot.

* SSRNode: Clean up.

* SSRNode: Honor material in `dispose()`.

linbingquan committed:
Examples: Clean up. (#29600)

sunag committed:
PostProcessingUtils: Improve names, replaced some `save` to `reset` (#29598)

* Improve names, replaced some `save` to `reset`

* Update SSRNode.js

Makio64 committed:
better error message

Makio64 committed:
better error message

Makio64 commented:
I was wondering if the error message shouldn't be factorize in function ? 
`validateInstanceId(instanceId, drawRanges)`
`validateGeometryId(geometryId, drawRanges)`

Like : 

```
function validateGeometryId(geometryId, drawRanges) {
  if (geometryId >= drawRanges.length || drawRanges[geometryId].active === false) {
    throw new Error(`BatchedMesh: Invalid geometryId ${geometryId}. Geometry is either out of range or has been deleted.`);
  }
}
```

Mugen87 committed:
PostProcessingUtils: Add `getScreenPosition()`. (#29599)

* PostProcessingUtils: Add `getScreenPosition()`.

* renamed types: `matrix4` -> `mat4`

---------

RenaudRohlinger committed:
WebGPURenderer: Fix missing updates to geometry attributes (#29589)

* WebGPURenderer: Fix geometry attribute not updated regardless of version

* BufferAttributes don't have uuid

* small cleanup

* more cleanup

* improve geometry comparison performance

* cleanup previous implementation

* attributes have no uuid

sunag committed:
Updated builds.

Mugen87 committed:
SSRNode: Honor `metalness`. (#29605)

sunag committed:
TSL: Introduce `premultipliedGaussianBlur` (#29604)

gkjohnson commented:
> I was wondering if the error message shouldn't be factorize in function ?
> validateInstanceId(instanceId, drawRanges)
> validateGeometryId(geometryId, drawRanges)

I think this is okay to do - it should simplify some of the functions

sunag committed:
Examples: Added `hashBlur` and improve `webgpu_backdrop_area` example (#29606)

* added hashBlur

* improve `webgpu_backdrop_area` example

* loop: fix generate const in `start` and `end`

* Update webgpu_backdrop_area.jpg

Makio64 committed:
TransformControls: Add min/max constraints. (#29602)

* add min/max constraints to TransformControls

* update TransformControls documentation

linbingquan committed:
Eslint: Add `no-irregular-whitespace` rule (#29608)

* Eslint: Add `no-irregular-whitespace` rule

* clean up

Mugen87 committed:
GaussianBlurNode: Remove useless check. (#29611)

Mugen87 committed:
ReflectorNode: Remove `getTextureNode()`. (#29609)

sunag committed:
TSL: `hashBlur` - added default `uv` (#29614)

* `hashBlur` - added default `uv`

* improve light intensity

Spiri0 committed:
WebGPURenderer: Introducing an `IndirectStorageBufferAttribute`  (#29594)

* Introducing an indirectStorageBuffer for drawIndirect

* Update NodeStorageBuffer.js

fixed lint issue

* Update StorageBufferNode.js

fixed lint issue

* Update IndirectStorageBufferAttribute.js

fixed lint issue

* Update StorageBufferNode.js

fixed lint issues

* Update StorageBufferNode.js

fixed lint issues

* Update StorageBufferNode.js

fixed lint issues

* Update StorageBufferNode.js

fixed lint issue

* PR update

* Update StorageBuffer.js

* Update Bindings.js

* Update Bindings.js

* Update StorageBufferNode.js

* rev

* cleanup

---------

Co-authored-by: Attila Schroeder <attila-schroeder.79@gmail.com>
Co-authored-by: sunag <sunagbrasil@gmail.com>

Mugen87 committed:
Update webgpu_postprocessing_ssr.html

Update info.

sunag committed:
WebGPURenderer: Introduce indirect drawing support (#29615)

* introduce indirect drawing support

* Update webgpu_particles.jpg

* update includes

* Update webgpu_particles.html

sunag committed:
Updated builds.

DevinCLane committed:
Typo: update prerequisites.html (#29616)

Small typo in prerequisites page.

> if you follow this rule **you** code will match...

switched to 

> if you follow this rule **your** code will match...

mrdoob committed:
Bumped servez.

dependabot[bot] committed:
Bump cookie and express (#29617)

Bumps [cookie](https://github.com/jshttp/cookie) and [express](https://github.com/expressjs/express). These dependencies needed to be updated together.

Updates `cookie` from 0.6.0 to 0.7.1
- [Release notes](https://github.com/jshttp/cookie/releases)
- [Commits](https://github.com/jshttp/cookie/compare/v0.6.0...v0.7.1)

Updates `express` from 4.21.0 to 4.21.1
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.1/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.21.0...4.21.1)

---
updated-dependencies:
- dependency-name: cookie
  dependency-type: indirect
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mrdoob committed:
Removed unused textures. (#29618)

mrdoob committed:
Clean up.

ShanHaiSiLu committed:
Docs: Improve `WebGLRenderer` Chinese page. (#29620)

Co-authored-by: ShanHaiJing <shanhaisilu@foxmail.com>

Mugen87 committed:
Update webgpu_postprocessing_ssaa.html

Fix typo.

sunag committed:
Nodes: Fixes and improvements for `reflector` and `gaussianBlur` (#29619)

* GaussianBlurNode: preserves input UV

* fix sequential events of update before

* add multi-sampler support

Mugen87 committed:
ReflectorNode: Add `getDepthNode()`. (#29621)

Co-authored-by: sunag <sunagbrasil@gmail.com>

sunag committed:
WaterMesh: Fix planar offset - WebGPU (#29626)

sunag committed:
ReflectorNode: Added missing `nodeObject` (#29627)

Mugen87 committed:
Updated builds.

aardgoose committed:
WebGPURenderer: webgl-fallback - fix uniform/texture index allocation (#29622)

* fix uniform index/texture use

* use explict test

* Update WebGLBackend.js

Minor code style clean up.

---------

Co-authored-by: aardgoose <angus.sawyer@email.com>
Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

Mugen87 committed:
ReflectorNode: Fix projection matrix computation for WebGL. (#29630)

Mugen87 committed:
Updated builds.

aardgoose committed:
remove useless call (#29631)

Co-authored-by: aardgoose <angus.sawyer@email.com>

Mugen87 committed:
Docs: Upgrade prettify.js. (#29633)

renovate[bot] committed:
Update actions/upload-artifact digest to b4b15b8 (#29638)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update actions/checkout digest to eef6144 (#29637)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency magic-string to v0.30.12 (#29641)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

puxiao committed:
Docs: Improve Chinese translation. (#29643)

renovate[bot] committed:
Update github/codeql-action digest to c36620d (#29640)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Mugen87 committed:
SSAAPassNode: Clean up. (#29647)

sungpaks committed:
Docs: Fix typo and Improve `How-to-dispose-of-objects` page Korean translation (#29649)

* Docs: Fix typo in Korean translation

* Docs: Update introduction `How-to-dispose-of-objects`

Mugen87 committed:
Update webgpu_postprocessing_dof.html

Ensure stats is updated.

hybridherbst committed:
Playground: Add `normalMap` node. (#29652)

Mugen87 committed:
WebGPUTextureUtils: Fix usage of `Texture.anisotropy`. (#29654)

sunag committed:
WebGPURenderer: Introduce `TiledLighting` (#29642)

* Renderer: move `.nodes.library` -> `.library`

* WGSLNodeBuilder: Remove incompatible types

* introduce lighting system

* introduce tiled lighting

* update imports

* cleanup

* revision

* `misc_controls_fly` : fix warning

* revision

* cleanup

* fix needsUpdate

sunag committed:
Updated builds.

donmccurdy committed:
fix(Material): sRGB decoding for VideoTexture .emissiveMap (#29657)

sawa-zen committed:
Update How-to-use-post-processing.html (#29665)

yomotsu committed:
fix: remove methods for CSSRenderer Objects within iframe (#29663)

* fix: remove methods for CSSRenderer Objects within iframe

* Element.remove() is widely supported

Mugen87 committed:
Addons: Add `TRAAPassNode`. (#29636)

* TRAAPass: Initial setup.

* TRAAPassNode: Implement clamping, fix velocity.

* TRAAPassNode: Finalize initial code.

* TRAAPassNode: Clean up.

* Examples: Clean up.

* TRAAPassNode: Clean up.

* TRAAPassNode: Fix dispose().

* TRAANodePass: Refactor MRT setup.

* TRAAPassNode: Clean up.

* TRAAPassNode: Make MRT handling more robust.

* TRAAPassNode: Implement prepass.

* Revert "TRAAPassNode: Implement prepass."

This reverts commit 172c2eeabc446facc53fb3f7f6dd134ba18e7009.

* TRAAPassNode: Clean up.

* WebGLBackend: Support render target textures in copyTextureToTexture().

* TRAAPassNode: Update comment.

* TRAAPassNode: Remove redundant clear().

gkjohnson committed:
Batchmesh: Fix cases where calling optimize can result in inconsistent state  (#29624)

* Fix skipping over active rather than inactive geometry

* Fix optimize

* Correct the shifting of geometry data

* Account for a case where the index bit size can be increased.

* typo fix

* Remove comments

* Add comments, removed unnecessary changes

* Dispose of textures on instance resize

zalo committed:
3MFLoader: Add Preliminary Support for Loading Implicit Functions (#29667)

* Begin adding an implicit function node parser

* Begin adding TSL Node Generation

There has to be a better way...

* Remove references to TSL

* Remove vestigial testing code

* Fix deep scan issue...

* Update 3MFLoader.js

Clean up.

---------

Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

jclaessens97 committed:
WebGPURenderer: Add backend information to `data-engine` field. (#29670)

* feat: add engine field to easier debug if running on webgl or webgpu

* Revert "feat: add engine field to easier debug if running on webgl or webgpu"

This reverts commit 95e506d50dcb72b3714f8ecba17918a8e713f38b.

* feat: add backend info to data-engine attribute instead

donmccurdy committed:
Examples: Clean up TSL editor example (#29669)

Mugen87 committed:
TRAAPassNode: Clean up. (#29671)

Mugen87 committed:
Examples: Clean up. (#29672)

Mugen87 committed:
Revert "WebGPURenderer: Add backend information to `data-engine` field. (#29670)" (#29675)

This reverts commit ac5804dd38b3bb4c43fec39c568658a754f00848.

gkjohnson committed:
WebGLRenderer: Correctly generate mipmaps for 3D & Array RenderTargets  (#29678)

* Enable generating mipmaps for 3d textures

* Update WebGLTextures.js

* Update WebGLTextures.js

donmccurdy committed:
GLTFExporter: Add support for WebGPURenderer (#29509)

* GLTFExporter: Add .setTextureUtils

* clean up

* Update misc_exporter_gltf.html

Fix import.

* fix missing await

* clean up

* GLTFExporter: Convert uncompatible int32 and uint32 attributes

---------

Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

Mugen87 committed:
WebGPUTextureUtils: Disable uv transform. (#29680)

Mugen87 committed:
WebGPUTextureUtils: Improve `uv` approach. (#29684)

Mugen87 committed:
Update webgl_loader_3mf.html

sunag committed:
WebGPURenderer: Fix pipeline cache key (#29685)

gkjohnson committed:
BatchedMesh: Consolidate internal geometry information  (#29687)

* Fix add instance bug

* Add todos for PR, instance count

* Progress towards replacing some geometry info

* remove remaining references to _drawRanges, _bounds, and _reservedRanges

* Fix rendering

* Remove unnecessary global fields

* Clean up

* Store draw range

* drawInfo -> instanceInfo

* Fix optimize

* revert "drawInfo" rename

* revert global changes to reduce noise

* Remove local variable name change to reduce noise

* remove accessors to reduce noise

Mugen87 committed:
AfterimagePass: Use copy shader. (#29691)

gkjohnson committed:
BatchedMesh: Batched cleanup 2 (#29695)

* Variable renaming, remove global unnecessary variables, add member variables

* drawInfo -> instanceInfo

* Don't initialize the geometry when resizing if it has not been initialized yet

* Fix copy-paste error

gkjohnson committed:
WebGLRenderer: Always generate mipmaps when "generateMipmaps" is true. (#29677)

* Always generate mipmaps when "generateMipmaps" is true.

* Update HTML examples

* More examples

* Update loaders

* Fix GLTF loader

* Update WebGPURenderer

* Remove unused constants

* Update bindings file to remove unnecessary mipmaps check

Mugen87 committed:
WebGLRenderer: Support render target textures in `copyTextureToTexture()`. (#29662)

* WebGLRenderer: Support render target textures in copyTextureToTexture().

* WebGLRenderer: Update `copyTextureToTexture3D()` and docs.

* WebGLRenderer: Attempt to fix `copyTextureToTexture3D()`.

Mugen87 committed:
WebGLTextures: Ensure depth texture properties are deleted. (#29697)

Mugen87 committed:
GLTFExporter: Better defaults for metallic and roughness factors. (#29692)

* GLTFExporter: Better defaults for metallic and roughness factors.

* GLTFExporter: Stick to glTF spec.

* GLTFExporter: Clean up.

gkjohnson committed:
WebGLRenderer: Copy set of layers when copying 3d texture data (#29699)

* Copy set of layers when copying 3d texture data

* Docs update

* Only use blit when copying depth texture data

sunag committed:
Examples: Cleanup `webgpu_lights_tiled`  (#29701)

sunag committed:
PhysicalLightingModel: Cleanup `evalIridescence` (#29702)

RenaudRohlinger committed:
WebGPURenderer: Introduce waitForGPU (#29686)

* WebGPURenderer: Introduce syncWithGPU

* cleanup

* rename method

* rename to waitForGPU

Mugen87 committed:
BufferGeometry: Check for existing attribute in `setFromPoints()`. (#29696)

* BufferGeometry: Check for existing attribute in `setFromPoints()`.

* BufferGeometry: Add warning to `setFromPoints()`.

* Update BufferGeometry.js

Fix typo.

* Docs: Improve `BufferGeometry.setFromPoints()`.

Mugen87 committed:
Updated builds.

Mugen87 committed:
Update PostProcessingUtils.js

Fix typo.

Mugen87 committed:
PostProcessingUtils: Add method for normal reconstruction. (#29703)

* PostProcessingUtils: Add method for normal reconstruction.

* DenoiseNode: Clean up.

* DenoiseNode: More clean up.

Mugen87 committed:
BloomNode: Remove redundant clears. (#29704)

renovate[bot] committed:
Update github/codeql-action digest to f779452 (#29706)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

donmccurdy committed:
TSL: Support defined color spaces in ColorSpaceNode (#29694)

* TSL: Support defined color spaces in ColorSpaceNode

* TSL: Add .convertColorNode()

* fix circular dependency

* remove chaining method

---------

sunag committed:
WebGPURenderer: Added double-side transmission 1/2 (#29705)

* added double-side transmission

* Update webgpu_materials_transmission.jpg

mrdoob committed:
GLTFExporter: Clearer code.

s-rigaud committed:
Docs: Fix typos (#29716)

Co-authored-by: Samuel Rigaud <srigaud@duodisplay.com>

linbingquan committed:
webgpu: add `wgsl` comment tag (#29709)

s-rigaud committed:
RectAreaLightTextureLib: Fix DataTexture usage (#29717)

Co-authored-by: Samuel Rigaud <srigaud@duodisplay.com>

Mugen87 committed:
Update OutlineNode.js

Fix white space.

Mugen87 committed:
Update OutlineNode.js

Fix parameter names.

Mugen87 committed:
Addons: Add `LensflareNode`. (#29715)

* LensflareNode: Experimenting with different approaches.

* LensflareNode: Improve example.

* Examples: Clean up

* Examples: Add comment.

* LensflareNode: Updating API.

* Examples: Clean up.

eXponenta committed:
WebGLRenderer: Stable reversed Z buffer implementation. (#29579)

* fix: stable reversed Z buffer implementation

Fix: reset clip state when reset is called
Fix: valid depth clear value when reversed is enabled

Feat:  non-persistent reversedZ state ( can be controlled via renderer.state.buffers.depth.setReversed(  )))

* fix: lint + misstakes
fix: program parameters

* Update WebGLCapabilities.js

---------

Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

Mugen87 committed:
Updated builds.

sunag committed:
`LensflareNode`,  `OutlineNode`: Added *Node suffix (#29719)

sunag committed:
WebGPURenderer: Added double-side transmission 2/2 (#29718)

* renders all back-side first in case of materials with transmission

* cleanup

snowiewdev committed:
Docs: added copy button to non-inline code snippets (#29700)

* docs: added copy button to non-inline code snippets

* docs: copy button icons img path update

sunag committed:
Examples: `webgpu_lights_tiled` cleanup (2) (#29722)

sunag committed:
TiledLightsNode: Fix light intensity (#29723)

* TiledLightsNode: Fix light intensity

* Update webgpu_lights_tiled.jpg

sunag committed:
WGSLNodeFunction: Ignore convertion of unknown types (#29724)

seopyoon committed:
fix: uuid translation in Korean (#29731)

gkjohnson committed:
ConeGeometry: Check for vertex convergence before removing triangles (#29728)

gkjohnson committed:
WebGLRenderer: Allow for copying 2d targets to and from layers of 3d textures (#29710)

* handle 2d and 3d cases in copyTextureToTexture3D

* Fix 3d check

* comments

* More comments

* Call 3d version from copyTextureToTexture

* Move texture properties back

* Adjust condition

* Deprecate copyTextureToTexture3D, update docs

* Add missing line to BatchedMesh

* Docs spelling

* Update docs

Mugen87 committed:
Addons: Deprecate MMD modules. (#29698)

* Addons: Deprecate MMD modules.

* MMD: Update link to new repo.

linbingquan committed:
PassNode: Clean up. (#29733)

sunag committed:
NodeMaterialObserver: Fix buffer resize when `transmission` is used (#29735)

sunag committed:
NodeMaterialObserver: Clean up (#29737)

linbingquan committed:
Examples: Clean up. (#29738)

Mugen87 committed:
Examples: Clean up. (#29743)

* Examples: Clean up.

* Examples: Clean up.

donmccurdy committed:
KTX2Loader: Support transcoding UASTC HDR to BC6H and RGBA16 (#29730)

* KTX2Loader: Support transcoding UASTC HDR to BC6H and RGBA16
* GLTFLoader: Fix incorrect texture.generateMipmaps setting

scottmas committed:
ScriptableNode: Rename `global` to `ScriptableNodeResources` (#29736)

* Rename problematic name for variable in ScriptableNode

* rename global to ScriptableResources

* Rename: ScriptableResources -> ScriptableNodeResources

---------

Co-authored-by: sunag <sunagbrasil@gmail.com>

Mugen87 committed:
ProgressiveLightMap: Remove usage of `THREE` namespace. (#29747)

* ProgressiveLightMap: Remove usage of `THREE` namespace.

* ProgressiveLightMap: Fix docs.

linbingquan committed:
Examples: Remove usage of `THREE` namespace. (#29748)

puxiao committed:
GLTFLoader: Add KTX2 mime type support in getImageURIMimeType (#29746)

Mugen87 committed:
ProgressiveLightMap: Add WebGPU version. (#29749)

* ProgressiveLightMapGPU: Initial commit.

* ProgressiveLightMap: Apply some fixes.

* ProgressiveLightMapGPU: More fixes.

* ProgressiveLightMap: Finalize implementation.

* E2E: Update screenshot.

aardgoose committed:
Addons: WebGPU CSM shadows - using shadowNode (#29610)

* CSM shadows - using shadowNode

* remove accidentally included file

* remove test logging

* Update CSMShadowNode.js

Fix main frustum with WebGL backend.

* rework

* handle camera changes

* CSMShadowNode: Clean up.

* CMSShadowNode: Fix cascade sampling.

* E2E: Update screenshot.

* CSMShadowNode: Clean up.

---------

Co-authored-by: aardgoose <angus.sawyer@email.com>
Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

Mugen87 committed:
Updated builds.

renovate[bot] committed:
Update actions/checkout digest to 11bd719 (#29754)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update actions/setup-node digest to 39370e3 (#29755)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mrxz committed:
WebXRManager: Inherit active layers from camera (#29742)

Mugen87 committed:
Addons: Remove `CinematicCamera`. (#29750)

* Addons: Remove `CinematicCamera`.

* E2E: Remove screenshot.

Mugen87 committed:
LDrawLoader: Add WebGPU support. (#29752)

* LDrawLoader: Add WebGPU support.

* LDrawLoader: Improve error handling.

* LDrawLoader: Clean up.

* Update Addons.js

Mugen87 commented:
Do you mind rebasing the PR?

Nokic233 committed:
Docs:Manual lights.html typo (#29756)

* Docs:Manual lights.html typo

* Docs:Manual lights.html code snippet intensity default 5

kristiker committed:
Add `Vector4.divide`, to match Vector3 and Vector2 function set (#29759)

Mugen87 committed:
NodeMaterial: Add support for `alphaHash`. (#29757)

* NodeMaterial: Add support for `alphaHash`.

* Clean up.

* E2E: Update exception list.

* Examples: Clean up.

* NodeMaterial: Fix alpha hash check.

Mugen87 committed:
Updated builds.

Mugen87 committed:
WebGPURenderer: Support `Scene.backgroundRotation`. (#29762)

sunag committed:
WebGPURenderer: `copyFramebufferToTexture` - support for post-rendering usage (#29729)

* support for post-rendering use

* cleanup

* Update Textures.js

* improve rectangle parameter values and types

* cleanup

Mugen87 committed:
RectAreaLightNode: Fix `update()`. (#29771)

Mugen87 committed:
TSL: Add function for BPCEM. (#29773)

* TSL: Add function for computing BPCEMs.

* E2E: Add example to exception list.

* added on resize

---------

Mugen87 committed:
Examples: Remove raymarching GLSL demo. (#29774)

Mugen87 committed:
Update SECURITY.md

Remove outdated link.

Mugen87 committed:
Update SECURITY.md

Fix typo.

RenaudRohlinger committed:
WebGPURenderer: Handle Device Lost Event (#29767)

* WebGPURenderer: Handle Device Lost Event

* small typo

* feedbacks

* fix

* not need to bind

mrdoob committed:
Updated package-lock.json

mrdoob committed:
Updated builds.

mrdoob committed:
r170

Mugen87 committed:
Dev: Upgrade jimp to latest version. (#29780)

Mugen87 committed:
Update constants.js

Update revision.

gkjohnson committed:
WebGLRenderer: Add support for copying mipmap data between textures (#29769)

* Add mipmap copy support

* unbind buffers

* working function implementation

* Support signature deprecation

* update docs

* Infer the mipmap sizes of the src texture

* Update WebGLRenderer.js

Update deprecation comment.

---------

Co-authored-by: Michael Herzog <michael.herzog@human-interactive.org>

RenaudRohlinger committed:
Materials: Revert static types of builtins materials (#29789)

* Materials: Revert static types of builtins materials

* git checkout materials too destructive

* git checkout materials too destructive

* adapt StandardNodeLibrary

* git checkout materials too destructive

Makio64 committed:
Merge branch 'fix-batchedmesh-silent-error' of https://github.com/Makio64/three.js into fix-batchedmesh-silent-error

Makio64 committed:
clean old

Makio64 commented:
closed in favor of : https://github.com/mrdoob/three.js/pull/29790
